src/pages/OrderPage.tsx

```diff
@@ -43,9 +55,15 @@ const OrderPage = () => {
       return;
     }
 
-    if (receiptCheckboxRef.current?.checked && !receiptNumberRef.current?.value.trim()) {
-      alert('현금영수증 번호를 입력해주세요.');
-      return;
+    if (receiptCheckboxRef.current?.checked) {
+      const receiptNumber = receiptNumberRef.current?.value.trim();
+      if (!receiptNumber) {
+        alert('현금영수증 번호를 입력해주세요.');
+        return;
+      } else if (!/^\d+$/.test(receiptNumber)) {
+        alert('현금영수증 번호는 숫자만 입력해주세요.');
+        return;
+      }
```

Comment:
elseif를 빼고 if-return로 통일하면 좀더 가독성이 좋아질 것 같아요, 어차피 조건문에 걸린 후 return처리하고 있기 때문에 동일효과입니다~

```suggestion
      if (!receiptNumber) {
        alert('현금영수증 번호를 입력해주세요.');
        return;
      }
      
       if (!/^\d+$/.test(receiptNumber)) {
        alert('현금영수증 번호는 숫자만 입력해주세요.');
        return;
      }
```